lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -128,6 +131,75 @@ class ReportUIFeatures {
     dropdown.addEventListener('click', this.onExport);
   }
 
+  _setupThirdPartyFilter() {
+    const pageTLDPlusOne = new URL(this.json.finalUrl).origin.split('.').slice(-2).join('.');
+
+    // get all tables with a text url
+    /** @type {Array<HTMLTableElement>} */
+    const tables = Array.from(this._document.querySelectorAll('.lh-table'));
+    tables.filter(el => el.querySelector('td.lh-table-column--url'))
+      .filter(el => /** @type {Element} */ (el.closest('.lh-audit')).id !== 'uses-rel-preconnect')
+      .forEach((el, index) => {
+        /** @type {NodeListOf<HTMLElement>} */
+        const urlItems = el.querySelectorAll('.lh-text__url');
+        /** @type {Record<string, HTMLTableRowElement>} */
+        const thirdPartyRows = {};
+
+        for (const urlItem of urlItems) {
+          const isThirdParty = !urlItem.title.includes(`${pageTLDPlusOne}/`);
+          if (!isThirdParty) {
+            continue;
+          }
+
+          const urlRow = urlItem.closest('tr');
+          if (urlRow) {
+            const rowPosition = Array.from(el.tBodies[0].children).indexOf(urlRow);
+            thirdPartyRows[`${rowPosition}`] = urlRow;
+          }
+        }
+
+        // create input box
+        const filterTemplate = this._dom.cloneTemplate('#tmpl-lh-3p-filter', this._document);
+        const filterInput = /** @type {HTMLInputElement} */ (filterTemplate.querySelector('input'));
+        const id = `lh-3p-filter-label--${index}`;
+
+        if (filterInput) {
+          filterInput.setAttribute('id', id);
+          filterInput.addEventListener('change', e => {
+            // remove elements from the dom and keep track of them to readd on uncheck
+            // why removing instead of hiding? nth-child(even) background-colors keep working
+            if (e.target instanceof HTMLInputElement && e.target.checked) {
+              Object.keys(thirdPartyRows).forEach(position => {
```

Comment:
`Object.entries(thirdPartyRows).forEach(([position, row]) => {` should help you get rid of the next line.